src/codegen.cpp

```diff
@@ -840,11 +841,13 @@ jl_llvm_functions_t jl_compile_linfo(jl_method_instance_t **pli, jl_code_info_t
     //         Get the codegen lock,
     //         And get the source
     if (li->def == NULL) {
+        JL_GC_PUSH1(&src);
```

Comment:
We are using a local array for the buffer so this has to be in the same or parent scope as the pop.